src/app/ui/components/containers/headers/dialog-header.tsx

```diff
@@ -22,13 +22,14 @@ export function DialogHeader({ onClose, title }: DialogHeaderProps) {
       width="100%"
       minHeight="40px"
     >
+      {title && onClose && <styled.div flex="1" />}
```

Comment:
Technically this works but we don't want to arbitrarily add empty `div` elements to help us style it. 

There are two cases we need to catch here:
1. when there is no title position the `HeaderActionButton` to the right
2. when there is a title
    - position the `HeaderActionButton` to the right
    - center the title text while accounting for the `HeaderActionButton` 
    
It could be better to restructure things and merge the `<Flex` for `title` and the `Box` for the button